src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
here we can also use some and reduce
```suggestion
  const women = withChildren
    ? people.filter((woman) => woman.sex === 'f' 
    && people.some((person) => person.mother === woman.name))
    : people.filter((woman) => woman.sex === 'f');

  const totalAge = women.reduce(
    (accumulator, woman) => accumulator + (woman.died - woman.born),
    0
  );

  return totalAge / women.length;
```